optimize/global.go

```diff
@@ -20,107 +20,116 @@ func DefaultSettingsGlobal() *Settings {
+	// composed to specify the valid fields. Methods are free to use or
 	// ignore these values.
 	//
-	// GlobalMethod may have its own specific convergence criteria, which can
+	// Method may have its own specific convergence criteria, which can
 	// be communicated using a MethodDone operation. This will trigger a
 	// PostIteration to be sent on result, and the MethodDone task will not be
-	// returned on result. The GlobalMethod must implement Statuser, and the
+	// returned on result. The Method must implement Statuser, and the
 	// call to Status must return a Status other than NotTerminated.
 	//
 	// The operation and result tasks are guaranteed to have a buffer length
-	// equal to the return from InitGlobal.
-	RunGlobal(operation chan<- GlobalTask, result <-chan GlobalTask, tasks []GlobalTask)
+	// equal to the return from Init.
+	Run(operation chan<- Task, result <-chan Task, tasks []Task)
 }
 
-// Global uses a global optimizer to search for the global minimum of a
-// function. A maximization problem can be transformed into a
-// minimization problem by multiplying the function by -1.
+// Minimize uses an optimizer to search for a minimum of a function. A
+// maximization problem can be transformed into a minimization problem by
+// multiplying the function by -1.
 //
 // The first argument represents the problem to be minimized. Its fields are
 // routines that evaluate the objective function, gradient, and other
 // quantities related to the problem. The objective function, p.Func, must not
 // be nil. The optimization method used may require other fields to be non-nil
-// as specified by method.Needs. Global will panic if these are not met. The
+// as specified by method.Needs. Minimize will panic if these are not met. The
 // method can be determined automatically from the supplied problem which is
 // described below.
 //
 // If p.Status is not nil, it is called before every evaluation. If the
 // returned Status is other than NotTerminated or if the error is not nil, the
 // optimization run is terminated.
 //
+// The second argument specifies the initial location for the optimization.
+// Some Methods do not require an initial location, but initX must still be
+// specified for the dimension of the optimization problem.
+//
 // The third argument contains the settings for the minimization. The
-// DefaultGlobalSettings function can be called for a Settings struct with the
-// default values initialized. If settings == nil, the default settings are used.
-// All of the settings will be followed, but many of them may be counterproductive
-// to use (such as GradientThreshold). Global cannot guarantee strict adherence
-// to the bounds specified when performing concurrent evaluations and updates.
+// DefaultSettingsLocal and DefaultSettingsGlobal functions can be called for
+// different default settings depending on the optimization method. If
+// settings is nil, the default settings are used. All settings will be honored
```

Comment:
perhaps mention that the default settings being then used are the DefaultSettingsLocal ?

Reply:
Done.